intl_translation_arb/test/extraction_test.dart

```diff
@@ -0,0 +1,487 @@
+    String invalidOuterGender(g) => Intl.gender(
+      g,
+      other: 'o',
+      desc: "Invalid outer gender",
+    );
+    ''', 'example.dart');
+
+      expect(extraction.warnings, [
+        '''
+Skipping invalid Intl.message invocation
+    <Intl.gender(g, other: 'o', desc: "Invalid outer gender")>
+    reason: The 'args' argument for Intl.message must be specified for messages with parameters. Consider using rewrite_intl_messages.dart
+    from example.dart    line: 1, column: 37'''
+      ]);
+    });
+
+    test('Gender message without name or args', () {
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String outerSelect(currency, amount) => Intl.select(
+          currency,
+          {
+            "CDN": "\$amount Canadian dollars",
+            "other": "\$amount some currency or other."
+          },
+          name: "outerSelect",
+          desc: "Select",
+          args: [currency, amount],
+        );
+    ''', 'example.dart');
+
+      final message = messages['outerSelect'];
+      expect(message.name, 'outerSelect');
+      expect(message.arguments, ['currency', 'amount']);
+      expect(icuMessageToString(message),
+          '{currency,select, CDN{{amount} Canadian dollars}other{{amount} some currency or other.}}');
+    });
+
+    test('Invalid select message', () {
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String failedSelect(currency) => Intl.select(
+      currency,
+      {"this.should.fail": "not valid", "other": "doesn't matter"},
+      name: "failedSelect",
+      args: [currency],
+      desc: "Invalid select",
+    );
+    ''', 'example.dart');
+      expect(extraction.warnings.first, '''
+Error IntlMessageExtractionException: Invalid select keyword: 'this.should.fail', must match '[a-zA-Z][a-zA-Z0-9_-]*'
+Processing <Intl.select(currency, {"this.should.fail" : "not valid", "other" : "doesn't matter"}, name: "failedSelect", args: [currency], desc: "Invalid select")>    from example.dart    line: 1, column: 38''');
+    });
+
+    test('Nested select and plural message without interpolation', () {
+      // A trivial nested plural/gender where both are done directly rather than
+      // in interpolations.
+      final extraction = MessageExtraction();
+      final messages = extraction.parseContent('''
+    String nestedOuter(number, gen) => Intl.plural(number,
+        other: Intl.gender(gen, male: "\$number male", other: "\$number other"),
+        name: 'nestedOuter',
+        args: [number, gen],
+        desc: "Gender inside plural");
+    ''', 'example.dart');
+
+      final message = messages['nestedOuter'];
+      expect(message.name, 'nestedOuter');
+      expect(message.arguments, ['number', 'gen']);
+      //print(icuMessageToString(message));
+      expect(icuMessageToString(message),
+          '{number,plural, other{{gen, select .....}}}');
+    });
+
```

Comment:
This test doesn't recognize the Intl.gender inside
icuMessageToString(message) returns {number,plural, other{}}

Reply:
Well that's interesting. That doesn't parse in intl_translation either, but it isn't noticed, because it doesn't fail parsing, it just ignores the interior. The ARB tests have that make_hardcoded_translation that writes out an ARB that does have that clause in it, so it works correctly in other languages. I think it would catch it if we used code generated from the ARB version for the original language, but we don't.

Obviously at some point I thought it could work. I think that when we examine the arguments, if one is a message and looks like an Intl message the code already exists. So that would be a nice thing if you could figure out how to fix it, but obviously as it doesn't work in the original, it's not essential.